src/drivers/imu/bmi088/BMI088_accel.cpp

```diff
@@ -0,0 +1,597 @@
+                                                                                     };
+
+BMI088_accel::BMI088_accel(int bus, const char *path_accel, uint32_t device, enum Rotation rotation) :
+    BMI088("BMI088_ACCEL", path_accel, bus, device, SPIDEV_MODE3, BMI088_BUS_SPEED, rotation),
+    ScheduledWorkItem(px4::device_bus_to_wq(get_device_id())),
+    _px4_accel(get_device_id(), (external() ? ORB_PRIO_MAX - 1 : ORB_PRIO_HIGH - 1), rotation),
+    _sample_perf(perf_alloc(PC_ELAPSED, "bmi088_accel_read")),
+    _measure_interval(perf_alloc(PC_INTERVAL, "bmi088_accel_measure_interval")),
+    _bad_transfers(perf_alloc(PC_COUNT, "bmi088_accel_bad_transfers")),
+    _bad_registers(perf_alloc(PC_COUNT, "bmi088_accel_bad_registers")),
+    _duplicates(perf_alloc(PC_COUNT, "bmi088_accel_duplicates")),
+    _got_duplicate(false)
+{
+    _px4_accel.set_device_type(DRV_ACC_DEVTYPE_BMI088);
+}
+
+BMI088_accel::~BMI088_accel()
+{
+    /* make sure we are truly inactive */
+    stop();
+
+    /* delete the perf counter */
+    perf_free(_sample_perf);
+    perf_free(_measure_interval);
+    perf_free(_bad_transfers);
+    perf_free(_bad_registers);
+    perf_free(_duplicates);
+}
+
+int
+BMI088_accel::init()
+{
+    /* do SPI init (and probe) first */
+    int ret = SPI::init();
+
+    /* if probe/setup failed, bail now */
+    if (ret != OK) {
+        DEVICE_DEBUG("SPI setup failed");
+        return ret;
+    }
+
+    return reset();
+}
+
+uint8_t
+BMI088_accel::read_reg(unsigned reg)
+{
+    // For the BMI088, you need to read out a dummy byte before you can read out the normal data (see section "SPI interface of accelerometer part" of the BMI088 datasheet)
+    uint8_t cmd[3] = { (uint8_t)(reg | DIR_READ), 0, 0};
+
+    transfer(cmd, cmd, sizeof(cmd));
+
+    return cmd[2]; // Skip dummy byte in cmd[1] and read out actual data
+}
+
+uint16_t
+BMI088_accel::read_reg16(unsigned reg)
+{
+    // For the BMI088, you need to read out the dummy byte before you can read out the normal data (see section "SPI interface of accelerometer part" of the BMI088 datasheet)
+    uint8_t cmd[4] = { (uint8_t)(reg | DIR_READ), 0, 0, 0 };
+
+    transfer(cmd, cmd, sizeof(cmd));
+
+    return (uint16_t)(cmd[2] << 8) | cmd[3]; // Skip dummy byte in cmd[1]
+}
+
+int BMI088_accel::reset()
+{
+    write_reg(BMI088_ACC_SOFTRESET, BMI088_SOFT_RESET); // Soft-reset
+    up_udelay(5000); // After a POR or soft-reset, the sensor needs up to 1ms boot time (see section "Power Modes: Accelerometer" in the BMI088 datasheet). Based off of testing it seems this value needs to be increased from 1ms. Setting it to 5ms.
+
+    // Perform a dummy read here to put the accelerometer part of the BMI088 back into SPI mode after the reset
+    // The dummy read basically pulls the chip select line low and then high
+    // See section "Serial Peripheral Interface (SPI)" of the BMI088 datasheet for more details.
+    read_reg(BMI088_ACC_CHIP_ID);
```

Comment:
According to the BMI088 datasheet I need to transition the CS pin from low to high of the accelerometer in order to switch it to SPI mode (It starts in I2C mode). Since I don't have access to the _dev pointer of the SPI device (it is declared as private in SPI.h), I can't call SPI_SELECT to just set the pin high and low. Therefore I do a dummy read operation which has the side effect of pulling the CS pin low and then high again after the read operation has completed. (See section "3. Quick Start Guide - Device Initialization").